src/main/java/emissary/core/channels/FillChannelFactory.java

```diff
@@ -70,18 +69,18 @@ private FillChannel(final long size, final byte value) {
         }
 
         @Override
-        protected long sizeImpl() throws IOException {
+        protected long sizeImpl() {
             return size;
         }
 
         @Override
-        protected void closeImpl() throws IOException {
+        protected void closeImpl() {
             // Nothing to do.
         }
 
         @Override
         protected int readImpl(final ByteBuffer byteBuffer, final int maxBytesToRead) throws IOException {
-            final int bytesToFill = Math.min(maxBytesToRead, Integer.MAX_VALUE);
+            final int bytesToFill = Math.min(maxBytesToRead, byteBuffer.remaining());
```

Comment:
Actual bug fix (with `Integer.MAX_VALUE` there this method is a no-op by definition), but I'm not sure it matters since I don't know any actual use case for `FillChannel` instances besides unit tests?